homeassistant/components/haveibeenpwned/sensor.py

```diff
@@ -148,43 +151,75 @@ def __init__(self, emails, api_key):
 
     def set_next_email(self):
         """Set the next email to be looked up."""
+        _LOGGER.debug("Setting the next email to look up email: %s", self._email)
         self._current_index = (self._current_index + 1) % self._email_count
         self._email = self._emails[self._current_index]
 
     def update_no_throttle(self):
         """Get the data for a specific email."""
+        _LOGGER.debug("Getting data for a specific email: %s", self._email)
         self.update(no_throttle=True)
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES, MIN_TIME_BETWEEN_FORCED_UPDATES)
     def update(self, **kwargs):
         """Get the latest data for current email from REST service."""
+        _LOGGER.debug(
+            "Getting the latest data for the current email from the REST service for email: %s",
+            self._email,
+        )
         try:
-            url = f"{URL}{self._email}?truncateResponse=false"
-            header = {USER_AGENT: HA_USER_AGENT, "hibp-api-key": self._api_key}
+            url = f"{URL}{self._email}"
+            paramspayload = {"truncateResponse": "false"}
+            header = {
+                "USER_AGENT": HA_USER_AGENT,
+                "hibp-api-key": self._api_key,
+            }
             _LOGGER.debug("Checking for breaches for email: %s", self._email)
-            req = requests.get(url, headers=header, allow_redirects=True, timeout=5)
+            req = requests.get(
+                url,
+                params=paramspayload,
+                headers=header,
+                allow_redirects=True,
+                timeout=5,
+            )
+            _LOGGER.debug("Requested URL is %s", url)
+            _LOGGER.debug("Request object URL is %s", req.request.url)
+            _LOGGER.debug("Request headers are %s", req.request.headers)
```

Comment:
All API-specific code has to be part of a third-party library hosted on PyPi. Home Assistant should only interact with objects and not make direct calls to the API.

We do not accept changes to existing code that pre-dates that rule. The integration needs to be refactored to adhere to our general design rules first.